spharpy/classes/coordinates.py

```diff
@@ -501,6 +590,27 @@ def _check_weights(self, weights):
 
         return weights
 
+    def _check_quadrature(self):
+        r"""Check if the sampling is a are valid quadrature.
+
+        Returns
+        -------
+        check : bool
+            Flag which indicates if quadrature is a valid quadrature
+        """
+        if self.n_max is None or self.weights is None:
+            return False
+        # generate SH object at N=1
+        n_eval = 1
+        sh_basis = spherical_harmonic_basis_real(n_eval, self)
+
+        # test if basis is quadrature
+        quad_evaluation = sh_basis.T @ np.diag(self.weights) @ sh_basis
+        identity = np.eye((n_eval + 1)**2)
+
+        error = np.abs(np.sum(np.sum(quad_evaluation-identity)))
```

Comment:
I would rather check for the max abs deviation. Currently positive and negative deviations could cancel out when taking the sum before the abs
```suggestion
        error = np.max(np.abs(quad_evaluation-identity))
```